frontends/p4/validateParsedProgram.cpp

```diff
@@ -65,14 +65,26 @@ void ValidateParsedProgram::container(const IR::IContainer* type) {
 }
 
 void ValidateParsedProgram::postorder(const IR::P4Table* t) {
-    auto ac = t->properties->getProperty(IR::TableProperties::actionsPropertyName);
+    auto ac = t->getActionList();
     if (ac == nullptr)
         ::error("Table %1% does not have an `%2%' property",
                 t->name, IR::TableProperties::actionsPropertyName);
-    auto da = t->properties->getProperty(IR::TableProperties::defaultActionPropertyName);
+    auto da = t->getDefaultAction();
     if (!isv1 && da == nullptr)
         ::warning("Table %1% does not have an `%2%' property",
                 t->name, IR::TableProperties::defaultActionPropertyName);
+
+    auto entriesList = t->getEntries();
+    if (entriesList != nullptr) {  // list of entries is optional
```

Comment:
Right. Actually this is a check that I forgot to remove and did not yet add to the type checker.